pkg/spec/spec_test.go

```diff
@@ -0,0 +1,44 @@
+package spec
+
+import (
+	"testing"
+
+	"github.com/pkg/errors"
+	"reflect"
+)
+
+func TestConflictingFields(t *testing.T) {
+	// This array should ideally go away and every struct in spec.go should
+	// get tested automatically, but for now, let's pass in all the structs
+	// https://stackoverflow.com/questions/20803758/how-to-get-all-defined-struct-in-golang
+	structsToTest := []interface{}{
+		&VolumeClaim{},
+		&ServicePortMod{},
+		&ServiceSpecMod{},
+		&IngressSpecMod{},
+		&EnvFromSource{},
+		&ConfigMapEnvSource{},
+		&Container{},
+		&ConfigMapMod{},
+		&PodSpecMod{},
+		&App{},
```

Comment:
Correct me if I am wrong, do we need to test everything? I think just testing `&App{}` will test everything recursively? Because in the end everything is embedded in it.